src/core/wee-command.c

```diff
@@ -5344,7 +5344,14 @@ COMMAND_CALLBACK(secure)
         }
         else
         {
+            if (temp_secure_passphrase == NULL || strcmp (temp_secure_passphrase, argv_eol[2]))
+            {
+                temp_secure_passphrase = strdup (argv_eol[2]);
```

Comment:
If `temp_secure_passphrase` was not `NULL` then it had some value that is different from `argv_eol[2]`. On this line, the pointer is overwritten and the previous `temp_secure_passphrase` string is not `free`-d, causing a leak.